crates/miniserve/src/lib.rs

```diff
@@ -66,21 +77,22 @@ impl Server {
     /// # Panics
     ///
     /// Panics if `127.0.0.1:3000` is not available.
-    pub fn run(self) {
-        let listener =
-            TcpListener::bind("127.0.0.1:3000").expect("Failed to connect to 127.0.0.1:3000");
+    pub async fn run(self) {
```

Comment:
Note that the `run` function is now asynchronous, indicated by the `async` keyword.